modules/electromagnetics/src/postprocessors/ReflectionCoefficient.C

```diff
@@ -0,0 +1,97 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ReflectionCoefficient.h"
+#include <complex>
+
+registerMooseObject("ElectromagneticsApp", ReflectionCoefficient);
+
+InputParameters
+ReflectionCoefficient::validParams()
+{
+  InputParameters params = SidePostprocessor::validParams();
+  params.addClassDescription(
+      "CURRENTLY ONLY FOR 1D PLANE WAVE SOLVES. Calculate power reflection coefficient "
+      "for impinging wave on a surface. Assumes that wave of form F = F_incoming + R*F_reflected");
+  params.addRequiredCoupledVar(
+      "field_real", "The name of the real field variable this postprocessor operates on.");
+  params.addRequiredCoupledVar("field_imag", "Coupled imaginary field variable.");
+  params.addRequiredParam<Real>("theta", "Wave incidence angle");
+  params.addRequiredParam<Real>("length", "Domain length");
+  params.addRequiredParam<Real>("k", "Wave number");
+  params.addParam<Real>("incoming_field_magnitude", 1.0, "Incoming field magnitude");
+  return params;
+}
+
+ReflectionCoefficient::ReflectionCoefficient(const InputParameters & parameters)
+  : SidePostprocessor(parameters),
+    MooseVariableInterface<Real>(this, false, "field_real"),
+    _qp(0),
+    _coupled_real(coupledValue("field_real")),
+    _coupled_imag(coupledValue("field_imag")),
+    _theta(getParam<Real>("theta")),
+    _length(getParam<Real>("length")),
+    _k(getParam<Real>("k")),
+    _incoming_mag(getParam<Real>("incoming_field_magnitude"))
+{
+}
+
+void
+ReflectionCoefficient::initialize()
+{
+  if (_current_elem->dim() > 1)
+  {
+    mooseError("The ReflectionCoefficient object is not currently configured to work with 2D or 3D "
+               "meshes. Please disable this object or setup a 1D mesh!");
+  }
+
+  _reflection_coefficient = 0;
+}
+
+void
+ReflectionCoefficient::execute()
+{
+  _reflection_coefficient = computeReflection();
+}
+
+PostprocessorValue
+ReflectionCoefficient::getValue()
+{
+  return _reflection_coefficient;
+}
+
+void
+ReflectionCoefficient::threadJoin(const UserObject & y)
+{
+  const ReflectionCoefficient & pps = static_cast<const ReflectionCoefficient &>(y);
+  Real temp_rc = _reflection_coefficient;
+  _reflection_coefficient = std::max(temp_rc, pps._reflection_coefficient);
+}
+
+void
+ReflectionCoefficient::finalize()
+{
+  gatherMax(_reflection_coefficient);
+}
+
+Real
+ReflectionCoefficient::computeReflection()
+{
+  std::complex<double> jay(0, 1);
+  std::complex<double> field(_coupled_real[_qp], _coupled_imag[_qp]);
+
+  std::complex<double> incoming_wave =
+      _incoming_mag * std::exp(jay * _k * _length * std::cos(_theta * libMesh::pi / 180.));
```

Comment:
jay...

I do think EM::j or EM::jj would kind of make sense? But then again maybe there will be other modules in the repo that will eventually use imaginary numbers...

Reply:
Also, why conversion to radians by default? I think maybe its best to let the arguments of std::cos be default and the user can convert units at the input file level. Just a comment...

Reply:
Good comment. Likely I left it this way due to the problem I was working in at the time, and neglected to correct it when I was trying to generalize things. It does make sense to leave things in the default arguments for cosine here and make the conversion in the input. 

Re: EM::j - I feel like I should just go with `EM::imaginary_unit` here to make things very obvious and readable. Then if other modules end up needing this, it'll be a quick move into the MOOSE namespace.

Reply:
EM::imaginary_unit is kind of wordy? I like EM::jay or EM::j better

Reply:
If at some point it needs to go into the framework, then we can re-name it and then do
```
namespace EM
{
using j = Moose::foo;
}

Reply:
@GiudGiud always the trouble right? Balancing clarity and wordiness 😄  I'll go with something and we can adjust later.